### PR TITLE
Movies

### PR DESCRIPTION
**Pull request by shapikso**

### CODE REVIEW

File: src/components/Movies/MovieCard/MovieCard.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { URL_IMG } from '../../../constants/api';
+import { Link } from 'react-router-dom';
+import './MovieCard.scss';
```
Comment (konstantinz1390) [inline src/components/Movies/MovieCard/MovieCard.js]: imports order

File: src/components/Movies/MovieCard/MovieCard.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import { formatRuntime, formatRate, getRateState } from '../../../helpers/format';
+
+
+const MovieCard = ({ id, backdrop_path, title, runtime, vote_average }) => {
+    const runtimeFormated = formatRuntime(runtime);
+    const formatedRate = formatRate(vote_average);
+    const typeRate = getRateState(vote_average);
+    return (
+        <Link to={`movie/${id}`}>
+            <div className="MovieCard card" data-movieid={id}>
```
Comment (konstantinz1390) [inline src/components/Movies/MovieCard/MovieCard.js]: different className style

File: src/components/Movies/MovieCard/MovieCard.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                    </div>
+                    <div className="film-description">
+                        <div className="film-name">
+                            <span>{title}</span>
+                        </div>
+                        <div className="film-duration">
+                            <span>{runtimeFormated}</span>
+                        </div>
+                    </div>
+                    <div className={"film-rate" + ' ' + `${typeRate}`}>
```
Comment (konstantinz1390) [inline src/components/Movies/MovieCard/MovieCard.js]: you can use only interpolation, omit concatenation

File: src/components/Movies/MovieCard/MovieCard.scss [hunk 4]
```diff
@@ -0,0 +1,94 @@
+@import '../../common/styles/colors.scss';
+
+.film-poster {
+    background: #000;
```
Comment (konstantinz1390) [inline src/components/Movies/MovieCard/MovieCard.scss]: use color const

File: src/components/Movies/MovieCard/MovieCard.scss [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    top: 10px;
+
+    justify-content: center;
+    align-items: center;
+    padding: 4px;
+    border-radius: 5px;
+    text-align: center;
+}
+.good-rate {
+    background-color: #246724;
```
Comment (konstantinz1390) [inline src/components/Movies/MovieCard/MovieCard.scss]: the same

File: src/components/Movies/Movies.js [hunk 6]
```diff
@@ -1,12 +1,73 @@
 import React, { Component } from 'react';
+import './Movies.scss';
```
Comment (konstantinz1390) [inline src/components/Movies/Movies.js:1]: imports order

File: src/components/common/Loader/Loader.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+
```
Comment (konstantinz1390) [inline src/components/common/Loader/Loader.js:3]: empty line

File: src/components/common/Loader/Loader.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+import './Loader.css';
```
Comment (konstantinz1390) [inline src/components/common/Loader/Loader.js]: imports order

File: src/components/common/Loader/Loader.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+const Loader = () => {
```
Comment (konstantinz1390) [inline src/components/common/Loader/Loader.js]: omit return if you don't have logic

File: src/components/common/Loader/Loader.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+    return (
+        <div className="loader">
+            <div className="lds-spinner">
+                <div></div>
```
Comment (konstantinz1390) [inline src/components/common/Loader/Loader.js]: suspicious implementation

File: src/components/Movies/MovieCard/MovieCard.js (hunk 3 above)
Comment (NataliaHuberna) [reply src/components/Movies/MovieCard/MovieCard.js]: `film-rate ${typeRate}`

File: src/components/Movies/MovieCard/MovieCard.js [hunk 11]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { URL_IMG } from '../../../constants/api';
```
Comment (nataliazhuravlova) [inline src/components/Movies/MovieCard/MovieCard.js:3]: .../

File: src/components/Movies/MovieCard/MovieCard.js (hunk 2 above)
Comment (nataliazhuravlova) [inline src/components/Movies/MovieCard/MovieCard.js]: data-movieid for what

File: src/components/Movies/MovieCard/MovieCard.scss [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+}
+.bad-rate {
+    background-color: #5B5B5B;
```
Comment (nataliazhuravlova) [inline src/components/Movies/MovieCard/MovieCard.scss]: the same

File: src/components/Movies/Movies.js [hunk 13]
```diff
@@ -1,12 +1,73 @@
...
 import Button from '../common/Button/Button';
+import MovieCard from './MovieCard/MovieCard';
+import axios from 'axios';
+import { URL_MOVIE, MOVIE_ON_PAGE } from '../../constants/api';
```
Comment (nataliazhuravlova) [inline src/components/Movies/Movies.js:6]: ../

File: src/components/Movies/Movies.js [hunk 14]
```diff
@@ -1,12 +1,73 @@
...
+    componentDidMount() {
+        this.getMovies();
+    }
+
+    getMovies = async () => {
+        try {
+            this.setState({ isLoading: true });
+            const { data } = await axios.get(`${URL_MOVIE}?page=${this.state.currentPage}&per_page=${MOVIE_ON_PAGE}`);
+            this.setState((prevState) => {
+                return {
```
Comment (nataliazhuravlova) [inline src/components/Movies/Movies.js]: without return

File: src/components/Movies/Movies.js [hunk 15]
```diff
@@ -1,12 +1,73 @@
...
+                            <MovieCard
+                                key={element.id}
+                                id={element.id}
+                                backdrop_path={element.backdrop_path} 
+                                title={element.title}
+                                runtime = { element.runtime } 
+                                vote_average= {element.vote_average} 
+                            />
+                        ))) : (
+                            <h1>You have no images!</h1>
```
Comment (nataliazhuravlova) [inline src/components/Movies/Movies.js]: images???

File: src/components/Movies/Movies.js [hunk 16]
```diff
@@ -1,12 +1,73 @@
...
+                        <Loader />
+                    ) : this.state.movies.length ? 
+                        (this.state.movies.map((element) => (
+                            <MovieCard
+                                key={element.id}
+                                id={element.id}
+                                backdrop_path={element.backdrop_path} 
+                                title={element.title}
+                                runtime = { element.runtime } 
+                                vote_average= {element.vote_average}
```
Comment (nataliazhuravlova) [inline src/components/Movies/Movies.js]: propaName in camelCase

File: src/components/Movies/Movies.js [hunk 17]
```diff
@@ -1,12 +1,73 @@
...
+                    {this.state.isLoading ? (
+                        <Loader />
+                    ) : this.state.movies.length ? 
+                        (this.state.movies.map((element) => (
+                            <MovieCard
+                                key={element.id}
+                                id={element.id}
+                                backdrop_path={element.backdrop_path} 
+                                title={element.title}
+                                runtime = { element.runtime }
```
Comment (nataliazhuravlova) [inline src/components/Movies/Movies.js]: without indent

File: src/components/Movies/Movies.scss [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+    height: 100%;
+    position: relative;
+}
+
+.movie-wrapper::after {
+    content: "";
+    width: 150px;
+    height: 150px;
+    border-radius: 50%;
+    background: linear-gradient(225.82deg, #352F44 24.16%, #5C5470 87.91%);
```
Comment (nataliazhuravlova) [inline src/components/Movies/Movies.scss]: colors

File: src/components/Movies/Movies.scss [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+    top: 50px;
+    left: -75px;
+    z-index: -1;
+}
+.movie-wrapper::before {
+    content: "";
+    width: 150px;
+    height: 150px;
+    border-radius: 50%;
+    background: linear-gradient(225.82deg, #352F44 24.16%, #5C5470 87.91%);
```
Comment (nataliazhuravlova) [inline src/components/Movies/Movies.scss]: the same

File: src/components/common/Loader/Loader.css [hunk 20]
```diff
@@ -0,0 +1,92 @@
...
+    display: inline-block;
+    position: relative;
+    width: 80px;
+    height: 80px;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
+}
+.lds-spinner div {
+  transform-origin: 40px 40px;
```
Comment (nataliazhuravlova) [inline src/components/common/Loader/Loader.css]: different indent in file

File: src/components/common/Loader/Loader.js (hunk 10 above)
Comment (nataliazhuravlova) [inline src/components/common/Loader/Loader.js]: use map

File: src/helpers/format.js [hunk 21]
```diff
@@ -0,0 +1,18 @@
...
+        return 'null';
+    const hours = Math.floor(runtime / 60);
+    const minutes = runtime % 60;
+    return `${hours}h ${minutes}m`;
+};
+
+export const formatRate = (rate) => (rate ? `${rate}` : NO_RATE);
+
+export const getRateState = (rate) => {
+    if (rate === null) return BAD_RATE;
```
Comment (nataliazhuravlova) [inline src/helpers/format.js]: !rate || !Number.isFinite(rateToNumber) || rateToNumber < 7) return BAD_RATE

File: src/components/common/Loader/Loader.scss [hunk 22]
```diff
@@ -0,0 +1,107 @@
...
+.lds-spinner div:after {
+  content: " ";
+  display: block;
+  position: absolute;
+  top: 3px;
+  left: 37px;
+  width: 6px;
+  height: 18px;
+  border-radius: 20%;
+  background: #fff;
```
Comment (konstantinz1390) [inline src/components/common/Loader/Loader.scss]: use const

**Pull request merged**